### ISSUE TITLE
SASRec model should use Binary cross entropy loss

### ISSUE THREAD

**Issue by asash** [RUCAIBox/RecBole#1667]
In the original paper the authors use Binary cross entropy as the loss function (relevant code in the original implementation: https://github.com/kang205/SASRec/blob/master/model.py#L88, which uses 1 random negative per positive)

In your code, you propose a choice from BPR and categorical cross-entropy
https://github.com/RUCAIBox/RecBole/blob/658cc90280f8ec76f99c14d3530823e593d3f7a4/recbole/model/sequential_recommender/sasrec.py#L116

While both of loss functions approaches are viable, they don't match the original implementation, and therefore this model can't be   called original SASRec. 

Could you please align your code with the original implementation?

**Comment by chenyuwuxin** [comment]
@asash  Hello, thanks for your enthusiastic attention!
This is our interface unification. We have implemented two kinds of loss for all seq class models: CE and BPR. The reason why we don't sample here is that we use CE loss instead of BCE loss; We currently do not implement the version of BCE loss for SASRec. This loss setting is mainly applicable to our data module. We will consider implementing BCE loss in our future development. Thank you again for your suggestions!

**Comment by asash** [comment]
Ok, but in that case i think it worth to clearly state in the documentation that your SASRec model != original SASRec. 
Many people use RECBole in academic research, as it provides a good selection of baselines. Unfortunately, if these baselines are not aligned with the original implementations, all these papers are flawed (i.e. maybe CE or BPR are not worse or even better than originally used BCE, but it makes it a different model, and using it instead of original SASRec is incorrect). 
Unification is good for production usage, but for the academic research it is very important for the model to match the original papers.
